### PR TITLE
code overhaul and move from YACL to MidnightLib

### PR DESCRIPTION
**Pull request by AmyMialee**

### CODE REVIEW

File: gradle.properties [hunk 1]
```diff
@@ -12,7 +12,7 @@ archives_base_name=miningutility
 # Dependencies
 # check this on https://modmuss50.me/fabric.html
 fabric_version=0.89.0+1.20.1
-yacl_version=3.2.1+1.20
+midnightlib_version=1.4.1-fabric
```
Comment (Luligabi1) [inline gradle.properties:15]: Please revert all config related changes - I'd rather use only YACL and not rely on 3rd party libraries for the core config functionality.

**Review by Luligabi1 — CHANGES_REQUESTED** [review]
Check comment above

**Comment by AmyMialee** [comment]
MidnightConfig.class is 1 class for the whole mod, you can copy the entire thing into your mod if you want (it's mentioned as a valid way to use it) and requires much less maintainance.
Especially when the old config system is like half the size of the whole mod.

**Comment by AmyMialee** [comment]
also is YACL not a 3rd party library?

**Comment by Luligabi1** [comment]
> also is YACL not a 3rd party library?

Yes, but it's only used for the config screen on ModMenu; if it stops being/takes longer to be updated, it's not holding back any releases.

> MidnightConfig.class is 1 class for the whole mod, you can copy the entire thing into your mod if you want (it's mentioned as a valid way to use it)
> Especially when the old config system is like half the size of the whole mod.

It's not a principle of size.

> and requires much less maintainance

This particular config system has been used on a lot of Cafeteria's mods for over 3 years now and it hasn't required any maintenance whatsoever on that time period.

**Pull request closed without merging**